### PR TITLE
Fix mono/osx tests

### PR DESCRIPTION
**Pull request by radical**

### CODE REVIEW

File: src/XMakeBuildEngine/UnitTests/Evaluation/Expander_Tests.cs [hunk 1]
```diff
@@ -3294,20 +3290,33 @@ public void Medley()
...
+#if FEATURE_WIN32_REGISTRY
+            if (NativeMethodsShared.IsWindows)
+            {
+                errorTests.Add("$(Registry:X)");
+            }
+            else
+            {
+                // If no registry, and not running on windows, this gets expanded to the empty string
+                // example: xplat build running on OSX
+                validTests.Add(new string[]{"$(Registry:X)", ""});
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/UnitTests/Evaluation/Expander_Tests.cs]: Should this be inside the `#if` guard? Or is it true for netcore and mono both?

File: src/XMakeTasks/UnitTests/GenerateResourceOutOfProc_Tests.cs [hunk 2]
```diff
@@ -2313,6 +2313,8 @@ public void Regress25163_OutputResourcesContainsInvalidPathCharacters()
     public class References
     {
         [Fact]
+        //FIXME: mono: looks for csc.exe
```
Comment (rainersigwald) [inline src/XMakeTasks/UnitTests/GenerateResourceOutOfProc_Tests.cs]: Can you file an MSBuild bug and put the URL here?

File: src/XMakeBuildEngine/UnitTests/Evaluation/Expander_Tests.cs (hunk 1 above)
Comment (radical) [inline src/XMakeBuildEngine/UnitTests/Evaluation/Expander_Tests.cs]: Hm the existing code tested this case only for !netcore. But this should probably be tested for netcore too. It seems to resolve to `""` even for netcore, so, I will update this.

**Comment by radical** [comment]
Updated. Thanks for the review!

**Comment by rainersigwald** [comment]
LGTM

**Comment by AndyGerlicher** [comment]
:shipit:

**Pull request merged**